src/trainer.py

```diff
@@ -1155,148 +1123,74 @@ def _save_checkpoint(self, training_state, phase="pretrain", new_best_macro=Fals
-            torch.save(
-                metric_info,
-                os.path.join(
-                    self._serialization_dir,
-                    task_name,
-                    "metric_state_target_train_epoch_{}{}.th".format(epoch, best_str),
-                ),
-            )
-            torch.save(
-                task_states[task_name],
-                os.path.join(
-                    self._serialization_dir,
-                    task_name,
-                    "task_state_target_train_epoch_{}{}.th".format(epoch, best_str),
-                ),
-            )
-            torch.save(
-                training_state,
-                os.path.join(
-                    self._serialization_dir,
-                    task_name,
-                    "training_state_target_train_epoch_{}{}.th".format(epoch, best_str),
-                ),
-            )
-
-            torch.save(
-                model_state,
-                os.path.join(
-                    self._serialization_dir,
-                    task_name,
-                    "model_state_target_train_epoch_{}{}.th".format(epoch, best_str),
-                ),
-            )
-            if new_best_macro:
-                self._unmark_previous_best("target_train", epoch, task=task_name)
-
-    def _find_last_checkpoint_suffix(self, search_phases_in_priority_order=["pretrain"]):
-
-        """
-        Search for checkpoints to load, looking only for `main` training checkpoints.
-        TODO: This is probably hairier than it needs to be. If you're good at string handling...
-        """
-        if not self._serialization_dir:
-            raise ConfigurationError(
-                "serialization_dir not specified - cannot "
-                "restore a model without a directory path."
-            )
-
-        for current_search_phase in search_phases_in_priority_order:
-            max_epoch = 0
-            to_return = None
-            candidate_files = glob.glob(
-                os.path.join(
-                    self._serialization_dir, "model_state_{}_*".format(current_search_phase)
-                )
-            )
-            for x in candidate_files:
-                epoch = int(
-                    x.split("model_state_{}_epoch_".format(current_search_phase))[-1].split(".")[0]
-                )
-                if epoch >= max_epoch:
-                    max_epoch = epoch
-                    to_return = x
-            return to_return.split("model_state_")[-1]
-
-    def _restore_checkpoint(self, search_phases_in_priority_order=["pretrain"]):
+    def _restore_checkpoint(self, phase, tasks=None):
         """
         Restores a model from a serialization_dir to the last saved checkpoint.
         This includes an epoch count and optimizer state, which is serialized separately
-        from  model parameters. This function should only be used to continue training -
+        from model parameters. This function should only be used to continue training -
```

Comment:
It'd help to say why: This will work, but it'll load a bunch of stuff you don't want.